rskj-core/src/main/java/co/rsk/peg/pegininstructions/PeginInstructionsBase.java

```diff
@@ -0,0 +1,71 @@
+package co.rsk.peg.pegininstructions;
+
+import co.rsk.core.RskAddress;
+import org.ethereum.util.ByteUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+public abstract class PeginInstructionsBase implements PeginInstructions {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeginInstructionsBase.class);
+    private final int protocolVersion;
+    protected RskAddress rskDestinationAddress;
+
+    protected PeginInstructionsBase(int protocolVersion) {
+        this.protocolVersion = protocolVersion;
+    }
+
+    private RskAddress getRskDestinationAddressFromData(byte[] data) {
+        byte[] rskDestinationAddressBytes = Arrays.copyOfRange(data, 2, 22);
+        return new RskAddress(rskDestinationAddressBytes);
+    }
+
+    protected abstract void validateDataLength(byte[] data) throws PeginInstructionsParseException;
+
+    protected abstract void parseAdditionalData(byte[] data) throws PeginInstructionsParseException;
+
+    public static int extractProtocolVersion(byte[] data) throws PeginInstructionsParseException {
+        if (data == null || data.length < 2) {
+            String message;
+
+            if (data == null) {
+                message = "Provided data is null";
+            }
+            else {
+                message = String.format("Invalid data given. Expected at least 2 bytes, " +
+                    "received %d", data.length);
+            }
+
+            logger.debug("[extractProtocolVersion] {}", message);
+            throw new PeginInstructionsParseException(message);
+        }
+
+        byte[] protocolVersionBytes = Arrays.copyOfRange(data, 0, 2);
+        return ByteUtil.byteArrayToInt(protocolVersionBytes);
+    }
+
+    @Override
+    public RskAddress getRskDestinationAddress() {
+        return this.rskDestinationAddress;
+    }
+
+    @Override
+    public int getProtocolVersion() {
+        return this.protocolVersion;
+    }
+
+    public void parse(byte[] data) throws PeginInstructionsParseException {
+        if (data.length < 22) {
+            logger.debug("[parse] Invalid data length");
+            String message = String.format("Invalid data length. Expected at least 22 bytes, "
+                + "received %d", data.length);
+            throw new PeginInstructionsParseException(message);
+        }
```

Comment:
Not sure if it makes sense to have this here, isn't the data length checked in `validateDataLength`?

Reply:
Length must be at least 22 bytes. Then, for v1, you check that length is 22 or 43 bytes, otherwise will fail. Maybe, for later versions, that validation will be different, but we always need at least those 22 bytes.

Reply:
Remove validation finally, each version will take care of validating the data length.